pkg/controller/elasticsearch/nodespec/podspec.go

```diff
@@ -100,6 +103,11 @@ func BuildPodTemplateSpec(
 		return corev1.PodTemplateSpec{}, err
 	}
 
+	// inherit the node labels list to trigger a restart when changed
+	if es.HasNodeLabelsToPodsAnnotations() {
+		builder = builder.WithAnnotations(map[string]string{esv1.NodeLabelsToPodsAnnotationsAnnotation: es.Annotations[esv1.NodeLabelsToPodsAnnotationsAnnotation]})
+	}
```

Comment:
Could we save an annotation here by incorporating the value of `esv1.NodeLabelsToPodsAnnotationsAnnotation` into the `config-hash`?

Reply:
The operator sets only one annotation on Elasticsearch Pods for now (`co.elastic.logs/module`) _(and it makes me wondering why we are using labels to store configuration hashes 🤔 )_.
But TBH this was my original intent, however the purpose of `elasticsearch.k8s.elastic.co/config-hash` is to store a sum of the configuration of the Elasticsearch node, my feeling is that it might be misleading to store something else. And I think it's the reason why we still have this TODO: https://github.com/elastic/cloud-on-k8s/blob/83b517e24593adee2108714acf6a5f0ec661f3c5/pkg/controller/elasticsearch/nodespec/podspec.go#L148-L152

Reply:
I guess if we use a wide definition of `config` it still could make sense to include keystore version or the node labels here, but I don't feel strongly about it.

Reply:
I implemented the idea in https://github.com/elastic/cloud-on-k8s/commit/c3e4a4719adca2e17277f4bb2fa9b81fa6b13c63. I have reused the same 32-bit FNV-1 hash function to have a "kind of consistent" hash format.